lib/sqs-handler.js

```diff
@@ -50,21 +59,69 @@ module.exports = class SQSHandler {
 
 		SQSEventStruct(event);
 
-		const consumer = new Consumer();
-		if(consumer.handlesBatch(event)) {
-			await consumer.processBatch(event.Records.map(record => {
-				const parsedRecord = this.parseRecord(record);
-				const logger = new LogTransport(record.messageId);
-				parsedRecord[Symbol.for('logger')] = logger;
-				return parsedRecord;
-			}));
-		} else {
-			await Promise.all(event.Records.map(record => {
-				const logger = new LogTransport(record.messageId);
-				return consumer.processSingleRecord(this.parseRecord(record), logger);
-			}));
+		const isBatch = (new Consumer()).handlesBatch(event);
+		if(isBatch)
+			await this.handleBatch(Consumer, event);
+		else
+			await this.handleSingle(Consumer, event);
+	}
+
+	/**
+	 * Process records in batch but splitting by client or without client
+	 *
+	 * @param {import('./sqs-consumer')} Consumer
+	 * @param {SQSEvent} event
+	 * @param {boolean} isBatch
+	 * @returns {Promise}
+	 */
+	static handleBatch(Consumer, event, isBatch = true) {
+		const recordsWithoutClient = [];
+		const recordsWithClient = {};
+		const consumerWithoutClient = new Consumer();
+		event.Records.forEach(record => {
+			const { preparedRecord, clientCode } = this.prepareRecord(consumerWithoutClient, record, isBatch);
+			if(clientCode) {
+				if(!recordsWithClient[clientCode])
+					recordsWithClient[clientCode] = [];
+
+				recordsWithClient[clientCode].push(preparedRecord);
+			} else
+				recordsWithoutClient.push(preparedRecord);
+		});
+
+		const batches = [];
+		if(recordsWithoutClient.length)
+			batches.push(consumerWithoutClient.processBatch(recordsWithoutClient));
+
+		if(Object.keys(recordsWithClient).length) {
+			Object.keys(recordsWithClient).forEach(clientCode => {
+				const consumer = new Consumer();
+				consumer.setSession({ clientCode });
+				batches.push(consumer.processBatch(recordsWithClient[clientCode]));
+			});
 		}
 
+		return Promise.all(batches);
+	}
+
+	/**
+	 * Process records one by one
+	 *
+	 * @param {import('./sqs-consumer')} Consumer
+	 * @param {SQSEvent} event
+	 * @param {boolean} isBatch
+	 * @returns {Promise}
```

Comment:
Solo para dejar en claro que el valor al que resuelve va a ser ignorado...

```suggestion
	 * @returns {Promise<void>}
```